add new module purchase_transport_document

lepistone opened the pull request:
This module adds a new object transport.document and allows to choose
those in the purchase order.
- [x] improve doc
- [x] add demo data
- [x] add security rule
- [x] demo -> data noupdate="1"
- [x] make name translatable
- [x] add POT

lepistone committed:
add new module purchase_transport_document

This module adds a new object transport.document and allows to choose
those in the purchase order.

yvaucher commented on purchase_transport_document/__openerp__.py:
```diff
@@ -0,0 +1,39 @@
+ "category": "Purchase Management",
+ "license": "AGPL-3",
+ 'complexity': "easy",
+ "description": """
```
You can create a real description and put it in a README.rst

lepistone replied:
you're right. I added a description and some demo data.

lepistone committed:
add security, doc, demo data

rdeheele commented:
:+1:

gurneyalex commented on purchase_transport_document/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+          "view/transport_document.xml",
+          "security/ir.model.access.csv",
+          ],
+ "demo": ["demo.xml"],
```
As the documents presented here are fairly standard, I'd set them in data.xml rather than demo.xml

gurneyalex commented on purchase_transport_document/model/transport_document.py:
```diff
@@ -0,0 +1,27 @@
+class TransportDocument(models.Model):
+    _name = "transport.document"
+
+    name = fields.Char('Name')
```
could you set this field as translatable? We probably want a POT file in i18n/ too

gurneyalex commented on purchase_transport_document/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+use the same documents.
+
+This way it is possible to specify which documents have to be supplied with the
+purchase. Possible examples (CMR, Bill of Langand and others) are provided as
```
Bill of what?? :astonished:

yvaucher replied:
Bill of Lading I guess :smile: 

BTW in 8.0 if you don't set a description in `__openerp__.py`, it will read a file name `README.rst`, `README.rst` or `README.txt` in module directory

https://github.com/odoo/odoo/blob/8.0/openerp/modules/module.py#L252

Thanks to @guewen for this tip

lepistone committed:
demo -> data

Those examples of transport documents are fairly common, so I always
include them (not only as demo data).

lepistone committed:
fix typo in manifest

lepistone committed:
make name translatable, add i18n template

lepistone commented:
@gurneyalex @yvaucher improved as you suggest. thanks!

yvaucher commented on purchase_transport_document/__openerp__.py:
```diff
@@ -0,0 +1,48 @@
+use the same documents.
+
+This way it is possible to specify which documents have to be supplied with the
+purchase. Possible examples (CMR, Bill of Landing and others) are provided as
```
Bill of lading not landing
http://en.wikipedia.org/wiki/Bill_of_lading

lepistone committed:
lading is a word!

lepistone commented:
@yvaucher fixed!

lepistone committed:
move doc to README.rst

yvaucher commented:
:+1: thanks

eLBati commented:
:+1:
